azure-pipelines/jobs/ci-linux.yml

```diff
@@ -57,18 +58,35 @@ jobs:
         )"
     continueOnError: true
 
+  # Files may be owned by root because builds don't set user ID. Later build steps run 'find' in
+  # the source tree, which fails due to permissions in the 'NetCore*-Int-Pool' queues. This step
+  # prevents the failure by using chown to clean up our source tree.
+  - script: |
+      set -x
+      docker run --rm \
+        -v "$(Agent.BuildDirectory):/root/build" \
+        -w /root/build \
+        bash -c "chown -R $(id -u):$(id -g) *"
+    displayName: Update file ownership from root to build agent account
+    continueOnError: true
+    condition: succeededOrFailed()
+
   # Create reference packages tarball and upload to azure storage
   - script: |
       set -x
       df -h
-      $(docker.run) $(docker.src.map) $(docker.src.work) $(imageName) ./publish.sh $(BUILD.BUILDNUMBER) --ci \
+      pushd $(rootDirectory)/source-build-reference-packages/src
+      ./publish.sh $(BUILD.BUILDNUMBER) --ci \
         /p:AzureAccountName=$(publish.blobStorage.account) \
-        /p:ContainerName=$(publish.blobStorage.container) \
         /p:AzureAccessToken=$(publish.blobStorage.accessToken) \
-        /p:ManifestName=$(BUILD.REPOSITORY.NAME) \
+        /p:ContainerName=$(publish.blobStorage.container) \
+        /p:DotNetPublishUsingPipelines=$(_PublishUsingPipelines) \
+        /p:DotNetArtifactsCategory=$(_DotNetArtifactsCategory) \
+        /p:ManifestName=source-build-reference-packages \
         /p:ManifestBuildId=$(BUILD.BUILDID) \
         /p:ManifestBranch=$(BUILD.SOURCEBRANCHNAME) \
         /p:ManifestCommit=$(BUILD.SOURCEVERSION)
+      popd
```

Comment:
`popd` shouldn't be necessary since this script just ends and state is lost. (You can use `cd`.)

In general though, I prefer using subshells `(...)` to handle this because the ending `)` can't be silently messed up, and you don't need to keep track and restore what changed. (It's not trivial to manually restore to the state before a `set -e`, for example.)